core/src/main/java/hudson/views/JobColumn.java

```diff
@@ -36,6 +39,24 @@
     public JobColumn() {
     }
 
+    public String columnCaption(View view) {
+        String metaName = null;
```

Comment:
I recommend to immediately return from the entry if there is no applicable <code>AlternativeUiTextProvider</code> implementations. Just a performance tweak